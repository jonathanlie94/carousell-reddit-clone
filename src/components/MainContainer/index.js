/*
  A container used to render components on the left side of the screen.
  Used by pages on containers.
*/

import React, { Component } from 'react';
import styled from 'styled-components';
import { theme } from 'ui';
import PropTypes from 'prop-types';

const Wrapper = styled.div`
  background-color: ${theme.colors.grey};
  color: ${theme.colors.black};
  height: auto;
  flex: 1;
  min-height: 80rem;
  padding: 12px 12px 96px 12px;
  position: relative;
`;

class MainContainer extends Component {
  render() {
    return (
      <Wrapper className={this.props.className}>
        {this.props.children}
      </Wrapper>
    );
  }
}

MainContainer.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node,
};

export default MainContainer;
