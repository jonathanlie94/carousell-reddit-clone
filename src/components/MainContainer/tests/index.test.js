import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import 'jest-styled-components';
import MainContainer from '../';

describe('<MainContainer />', () => {
  it('should render children when passed in', () => {
    const renderedComponent = shallow(
      <MainContainer>
        <div className="unique-class-name" />
      </MainContainer>
    );
    expect(
      renderedComponent.contains(<div className="unique-class-name" />)
    ).toEqual(true);
  });

  it('should match previous snapshot', () => {
    const renderedComponent = shallow(<MainContainer  />);
    const tree = toJSON(renderedComponent);
    expect(tree).toMatchStyledComponentsSnapshot();
  });
});
