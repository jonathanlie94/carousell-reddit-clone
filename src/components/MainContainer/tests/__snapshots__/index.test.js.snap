// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainContainer /> should match previous snapshot 1`] = `<styled.div />`;
